integrations-service/integrations/utils/integrations/mailgun.py

```diff
@@ -0,0 +1,63 @@
+import aiohttp
+from beartype import beartype
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from ...autogen.Tools import MailgunSendEmailArguments, MailgunSetup
+from ...env import mailgun_api_key  # Import env to access environment variables
+from ...models import MailgunSendEmailOutput
+
+
+@beartype
+@retry(
+    wait=wait_exponential(multiplier=1, min=4, max=10),
+    reraise=True,
+    stop=stop_after_attempt(3),
+)
+async def send_email(
+    setup: MailgunSetup, arguments: MailgunSendEmailArguments
+) -> MailgunSendEmailOutput:
+    """
+    Sends an email with the provided details using the Mailgun API.
+    """
+
+    try:
+        domain = arguments.from_.split("@")[-1]
+    except Exception as e:
+        msg = f"Error splitting email domain: {e}"
+        raise Exception(msg)
+
+    # Extract domain from from_ address (e.g. "test@example.com" -> "email.example.com")
+    if not domain.startswith("email."):
```

Comment:
why do we need this?

Reply:
It is possible that the email is written in this form `test@email.example.com` (it was the case in one of the cookbooks), so I thought we should allow both ways. wdyt?

Reply:
yes it is